DCOS/templates/preprovision/extensions/preprovision-agent-windows/v1/preprovision-agent-windows.ps1

```diff
@@ -1,13 +1,28 @@
-$MESOS_SERVICE_DIR = Join-Path $env:SystemDrive "DCOS\mesos\service"
-New-Item -ItemType Directory -Force -Path $MESOS_SERVICE_DIR
+$ErrorActionPreference = "Stop"
 
-$UTF8NoBOM = New-Object System.Text.UTF8Encoding $False
+$BASE_URL = "http://dcos-win.westus.cloudapp.azure.com/dcos-windows/testing/preprovision/extensions/preprovision-agent-windows/v1"
+$SCRIPTS_DIRECTORY = Join-Path $env:TEMP "preprovision_scripts"
+$AGENT_CREDS_SCRIPT="preprovision-agent-windows-credentials.ps1"
+$FLUENTD_SCRIPT="preprovision-agent-windows-fluentd.ps1"
+$SCRIPTS = @{
+    "agent_creds" = @{
+        "url" = "$BASE_URL/$AGENT_CREDS_SCRIPT"
+        "local_file" = Join-Path $SCRIPTS_DIRECTORY $AGENT_CREDS_SCRIPT
+    }
+	"fluentd" = @{
+        "url" = "$BASE_URL/$FLUENTD_SCRIPT"
+        "local_file" = Join-Path $SCRIPTS_DIRECTORY $FLUENTD_SCRIPT
+    }
+}
 
-$cred = "{`"principal`": `"mycred1`", `"secret`": `"mysecret1`"}"
-[System.IO.File]::WriteAllLines("$MESOS_SERVICE_DIR\credential.json", $cred, $UTF8NoBOM)
+if(!(Test-Path $SCRIPTS_DIRECTORY)) {
+        New-Item -ItemType "Directory" -Path $SCRIPTS_DIRECTORY
+}
 
-$httpCred = "{`"credentials`": [{`"principal`": `"mycred2`", `"secret`": `"mysecret2`"}]}"
-[System.IO.File]::WriteAllLines("$MESOS_SERVICE_DIR\http_credential.json", $httpCred, $UTF8NoBOM)
-
-$serviceEnv = "MESOS_AUTHENTICATE_HTTP_READONLY=true`r`nMESOS_AUTHENTICATE_HTTP_READWRITE=true`r`nMESOS_HTTP_CREDENTIALS=$MESOS_SERVICE_DIR\http_credential.json`r`nMESOS_CREDENTIAL=$MESOS_SERVICE_DIR\credential.json"
-[System.IO.File]::WriteAllLines("$MESOS_SERVICE_DIR\environment-file", $serviceEnv, $UTF8NoBOM)
+foreach($script in $SCRIPTS.Keys) {
+	Write-Host "Downloading: $($SCRIPTS[$script]["url"])"
+	Start-BitsTransfer $SCRIPTS[$script]["url"] -Destination $SCRIPTS[$script]["local_file"]
```

Comment:
Don't use `Start-BitsTransfer` as it might fail when executing via remote calls (ex: WinRM). Please use `Invoke-WebRequest -UseBasicParsing -Uri <url> -OutFile <local_file_path>` just like [here](https://github.com/Microsoft/mesos-jenkins/blob/master/Mesos/start-windows-build.ps1#L86).